### PR TITLE
Add C++ Solution + Editorial to Fort Moo (Plat)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
2016 plat moment

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-600.mdx [hunk 1]
```diff
@@ -2,19 +2,105 @@
...
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
 
+## Solution
+
+### Explanation
+
+By computing the [2D Prefix Sum](/silver/more-prefix-sums/) for the grid, we can evaluate the usability of a frame in $\mathcal{O}(1)$ time.
```
Comment (andrewwangva) [inline solutions/usaco-600.mdx:14]: How is this so much different than the official?

Comment (SansPapyrus683) [reply solutions/usaco-600.mdx:14]: even if it isn't, adding an internal solution does no harm

File: solutions/usaco-600.mdx [hunk 2]
```diff
@@ -2,19 +2,105 @@
...
+
+int main() {
+	freopen("fortmoo.in", "r", stdin);
+	freopen("fortmoo.in", "w", stdout);
+
+	int n, m;
+	cin >> n >> m;
+
+	vector<vector<bool>> grid(n, vector<bool>(m, false));
+	vector<vector<int>> cs(n + 1, vector<int>(m + 1, 0));
```
Comment (SansPapyrus683) [inline solutions/usaco-600.mdx:36]: wth is `cs`
also remove the 0 and false in the ector declaration

File: solutions/usaco-600.mdx [hunk 3]
```diff
@@ -2,19 +2,105 @@
...
+		for (int j = 0; j < m; j++) {
+			grid[i][j] = s[j] == 'X';
+
+			// build prefix sum
+			cs[i + 1][j + 1] = cs[i][j + 1] + cs[i + 1][j] - cs[i][j];
+			cs[i + 1][j + 1] += grid[i][j];
+		}
+	}
+
+	int ans = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-600.mdx:51]: (optional) rename to `valid_forts`

File: solutions/usaco-600.mdx [hunk 4]
```diff
@@ -2,19 +2,105 @@
...
+
+	for (int c1 = 0; c1 < m; c1++) {
+		for (int c2 = 0; c2 < m; c2++) {
+			int prev = -1;
+
+			for (int r = 0; r < n; r++) {
+				bool emp = (cs[r + 1][c2 + 1] - cs[r][c2 + 1] - cs[r + 1][c1] + cs[r][c1]) == 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-600.mdx:58]: 80 chars?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-600.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-600.mdx:36]: cumulative sum

File: solutions/usaco-600.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-600.mdx:14]: + it implements using a diff lang

File: solutions/usaco-600.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/usaco-600.mdx:51]: ill rename to `max_area`

File: solutions/usaco-600.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/usaco-600.mdx:58]: image

Comment (SansPapyrus683) [reply solutions/usaco-600.mdx]: But does that include indentation?

File: solutions/usaco-600.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-600.mdx]: rename to pref sim maybe?

Comment (ryanchou-dev) [reply solutions/usaco-600.mdx]: k

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-600.mdx [hunk 5]
```diff
@@ -2,19 +2,106 @@
...
+
+	for (int i = 0; i < n; i++) {
+		string s;
+		cin >> s;
+
+		for (int j = 0; j < m; j++) {
+			grid[i][j] = s[j] == 'X';
+
+			// build prefix sum
+			pref_sum[i + 1][j + 1] = pref_sum[i][j + 1] + pref_sum[i + 1][j] - pref_sum[i][j];
```
Comment (SansPapyrus683) [inline solutions/usaco-600.mdx:46]: over 80

**Review by SansPapyrus683 — APPROVED** [review]

**Review by TruaShamu — APPROVED** [review]
Looks good to me

**Comment by sathu-beep** [comment]
why do you have spaces between lines that dont need spaces?

**Comment by SansPapyrus683** [comment]
what

**Comment by ar492** [comment]
> why do you have spaces between lines that dont need spaces?

this is fine

**Review by ar492 — APPROVED** [review]
i trust the webdev portions are fine

**Comment by sathu-beep** [comment]
thanks
Thanks,
Sathvik

**Pull request merged**